handler_creator.go

```diff
@@ -25,27 +22,30 @@ func handlerByMethod(route *string, do interface{}, method string) *handler {
 			h.headerOffset = i
 		default:
 			// We not sure what is it, so just assume it's user defined context
-			h.userDefinedContextOffset = i
+			h.userContextsOffset = append(h.userContextsOffset, i)
```

Comment:
Seems we should extract out user context code to have an independent part to make sure these crazy code won’t keeping growing up

Reply:
@dannypsnl Do you think we should export a context package to handle fill context & generate context offset?

Reply:
Yes, better to put it under `internal` pkg